addons/payment_ogone/controllers/main.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import pprint
+
+import werkzeug
+
+from odoo import _, http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class OgoneController(http.Controller):
+    _accept_url = '/payment/ogone/accept'
+    _decline_url = '/payment/ogone/ogonedecline'
+    _exception_url = '/payment/ogone/exception'
+    _cancel_url = '/payment/ogone/cancel'
+    _fleckcheckout_url = '/payment/ogone/flexchekout/feedback'
+    _fleckcheckout_final_url = "/payment/ogone/flexchekout/final"
+
+    @http.route('/payment/ogone/payment_setup', type='json', auth='public')
+    def payment_setup(self, acquirer_id, amount=None, currency_id=None, partner_id=None, **data):
```

Comment:
```suggestion
    def ogone_payment_setup(self, acquirer_id, amount=None, currency_id=None, partner_id=None, **data):
```
Routes' methods must have a unique name to avoid being overwritten by any other module.